Integration test for revert reason detection w/ ganache 2.1.0

remon-nashid opened the pull request:
This is a followup to #8. Apologies for the late submission.

This PR brings an integration test that does the following:
- Deliberately installs an older version of ganache-cli that, in turn, depends on ganache-core < 2.2.0. 
- Assert that `shouldFail.withMessage` accommodates older versions of ganache by skipping reason message check.

nventuro commented on test-integration/ganache-pre-2.2/test/Tested.test.js:
```diff
@@ -0,0 +1,31 @@
+contract('Tested', function (accounts) {
+  context('shouldFail.reverting.withMessage', async function () {
+    beforeEach(async function () {
+      this.contract = await Tested.new(0);
```
What is this 0? `Tested` doesn't seem to have a constructor.

remon-nashid replied:
Ouch that's a copy-n-paste mistake. Will remote it.

nventuro commented:
Have you tried running this test manually? Can the error message be clearly seen in the Mocha output?

remon-nashid commented:
> Have you tried running this test manually? Can the error message be clearly seen in the Mocha output?

Yes I do see the warning in test output, but I admit it can be harder to spot amongst mocha's pretty output. Please see attached integration test output.
[output.txt](https://github.com/OpenZeppelin/openzeppelin-test-helpers/files/2882816/output.txt)

remon-nashid commented:
I've addressed review issues and colored warning message in hope it becomes more noticeable. Here is how it looks in Travis:

![warning](https://user-images.githubusercontent.com/1994818/53066475-1008cc80-349e-11e9-88fd-4d65b1cbf46e.png)

nventuro commented on .travis.yml:
```diff
@@ -21,8 +21,8 @@ jobs:
-      name: "Truffle 5.x"
-      script: ./test-integration/simple-project-truffle-5.x/run.sh
+      name: "Integration tests"
+      script: npm run test-integration
```
Hm, I had set it up so that each test was a single job, allowing us to easily see that they were all being run, passing, etc. Something like

```
- stage: Integration tests
  name: "Truffle 5.x"
  script: ./test-integration/simple-project-truffle-5.x/run.sh

- stage: Integration tests
  name: Ganache 2.2.x
  script: ./test-integration/ganache-2.2.x/run.sh
```

nventuro commented on src/shouldFail.js:
```diff
@@ -1,4 +1,5 @@
 const { should } = require('./setup');
+const c = require('ansi-colors');
```
I know this is what they suggest, but having `c` be a variable containing a module isn't the best idea IMO. Could we rename this to something like `colors`?

nventuro commented:
Great work @remon-nashid, thanks a lot!
